PyRDF/Node.py

```diff
@@ -61,6 +61,40 @@ def __init__(self, get_head, operation):
         self._cur_attr = "" # Name of the new incoming operation
         self.value = None
 
+    def __getstate__(self):
+        """
+        Converts the state of the current node
+        to a Python dictionary.
+
+        Returns
+        -------
+        Python Dictionary
+            A dictionary that stores all instance variables
+            that represent the current PyRDF node.
+
+        """
+        state_dict = {'children':self.children}
+        if not self.operation:
+            return state_dict
+
+        state_dict['operation_name'] = self.operation.name
```

Comment:
I also would say that maybe you could do something like `state_dict.update(self.operation. __dict__)`, no?
It add an extra field, but it would be cleaner.

Reply:
@diocas , `operation.__dict__` would also have the Enum variable named `Types`. It is not needed while pickling right ?

Reply:
That's why I said that it was *only* one extra field. But I forgot it was an Enum, so forget this suggestion. As mixing Enums and pickle is not a good idea..

Reply:
Is this fine @diocas ?
```python
dict = self.operation.__dict__
dict.pop('op_type')
state_dict.update(dict)
```

Reply:
meh
I don't really like this one.. keep it as is

Reply:
Ok 👍